src/server.cc

```diff
@@ -201,15 +201,17 @@ InferenceServer::Init()
     ready_state_ = ServerReadyState::SERVER_FAILED_TO_INITIALIZE;
     return status;
   }
-
-  PinnedMemoryManager::Options options(pinned_memory_pool_size_);
-  status = PinnedMemoryManager::Create(options);
-  if (!status.IsOk()) {
-    ready_state_ = ServerReadyState::SERVER_FAILED_TO_INITIALIZE;
-    return status;
+  if (pinned_memory_pool_size_ > 0) {
+    PinnedMemoryManager::Options options(pinned_memory_pool_size_);
```

Comment:
I think we have host policy options as well. @GuanLuo do you know if it is possible to have host policy configs that require allocating pinned memory even when pinned memory pool size is zero.

Reply:
It looks like this new `if (pinned_memory_pool_size)` is causing issues when pinned memory pool size is set to zero (`--pinned-memory-pool-byte-size=0`).

This only happens to be used in 2 tests:
- L0_pinned_memory : segfaults
- L0_metrics : OK because it doesn't do any inference with this setting, it just checks the metrics

For some flows (possibly most?), when trying to allocate an output buffer it will try pinned -> cpu fallback order within PinnedMemoryManager, but it looks like never gets to cpu fallback because it returns earlier due to no PinnedMemoryManager instance existing.

Reply:
@Tabrizian I looked at the internals of PinnedMemoryManager
Should we move the check inside of PinnedMemoryManager::Create(const Options& options)
That way we just avoid the cuda calls and create a PinnedMemoryManager

Reply:
+1 

For now, we need the PinnedMemoryManager even for non-pinned CPU memory allocations. It is a simpler WAR to always have the manager, and moved the pinned_memory size checks inside of the manager to avoid unnecessary CUDA calls.